### ISSUE TITLE
Gives (mostly) useless results for structs nested within classes

### ISSUE THREAD

**Issue by auscompgeek** [robotpy/robotpy-cppheaderparser#11]
Encountered when I was working on robotpy-rev. Struct in question is `rev::CANSparkMax::ExternalFollower`.

* The struct gets listed as a class at the header level, as well as in the class nested classes list.
* Any method parameters that use the nested struct as its type are "unresolved".

**Comment by virtuald** [comment]
Yeah. There's a lot of badness related to namespaces and scopes. It's a wonder this thing works at all.
